app/repositories/EventReportCacheRepository.scala

```diff
@@ -0,0 +1,219 @@
+import models.enumeration.ApiTypes
+import org.joda.time.{DateTime, DateTimeZone}
+import org.mongodb.scala.bson.BsonBinary
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model._
+import play.api.libs.json._
+import play.api.{Configuration, Logging}
+import repositories.EventReportCacheEntry.EventReportCacheEntryFormats.{apiTypesKey, expireAtKey, pstrKey}
+import repositories.EventReportCacheEntry.{DataEntry, EventReportCacheEntry, EventReportCacheEntryFormats, JsonDataEntry}
+import uk.gov.hmrc.crypto.{Crypted, CryptoWithKeysFromConfig, PlainText}
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.formats.MongoBinaryFormats.{byteArrayReads, byteArrayWrites}
+import uk.gov.hmrc.mongo.play.json.formats.MongoJodaFormats
+import uk.gov.hmrc.mongo.play.json.{Codecs, PlayMongoRepository}
+
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.TimeUnit
+import scala.concurrent.{ExecutionContext, Future}
+
+
+object EventReportCacheEntry {
+
+  sealed trait EventReportCacheEntry
+
+  case class DataEntry(pstr: String, apiTypes: String, data: BsonBinary, lastUpdated: DateTime, expireAt: DateTime)
+    extends EventReportCacheEntry
+
+  case class JsonDataEntry(pstr: String, apiTypes: String, data: JsValue, lastUpdated: DateTime, expireAt: DateTime)
+    extends EventReportCacheEntry
+
+  object DataEntry {
+    def apply(pstr: String,
+              apiTypes: ApiTypes,
+              data: Array[Byte],
+              lastUpdated: DateTime = DateTime.now(DateTimeZone.UTC),
+              expireAt: DateTime): DataEntry = {
+
+      DataEntry(pstr, apiTypes.toString, BsonBinary(data), lastUpdated, expireAt)
+    }
+
+    final val bsonBinaryReads: Reads[BsonBinary] = byteArrayReads.map(BsonBinary(_))
+    final val bsonBinaryWrites: Writes[BsonBinary] = byteArrayWrites.contramap(_.getData)
+    implicit val bsonBinaryFormat: Format[BsonBinary] = Format(bsonBinaryReads, bsonBinaryWrites)
+
+    implicit val dateFormat: Format[DateTime] = MongoJodaFormats.dateTimeFormat
+    implicit val format: Format[DataEntry] = Json.format[DataEntry]
+  }
+
+  object JsonDataEntry {
+    def applyJsonDataEntry(pstr: String,
+                           apiTypes: ApiTypes,
+                           data: JsValue,
+                           lastUpdated: DateTime = DateTime.now(DateTimeZone.UTC),
+                           expireAt: DateTime): JsonDataEntry = {
+
+      JsonDataEntry(pstr, apiTypes.toString, data, lastUpdated, expireAt)
+    }
+
+    implicit val dateFormat: Format[DateTime] = MongoJodaFormats.dateTimeFormat
+    implicit val format: Format[JsonDataEntry] = Json.format[JsonDataEntry]
+  }
+
+  object EventReportCacheEntryFormats {
+    implicit val dateFormat: Format[DateTime] = MongoJodaFormats.dateTimeFormat
+    implicit val format: Format[EventReportCacheEntry] = Json.format[EventReportCacheEntry]
+
+    val pstrKey = "pstr"
+    val inviteePsaIdKey = "inviteePsaId"
```

Comment:
Do we need this?

Reply:
I think there is no cache requirement so far for this ticket

Reply:
The usage of format is on 105 -> 108 as ERCacheRepo extends PlayMongoRepo, when defining domainFormat and extraCodecs.

Reply:
The inviteePsaIdKey isn't used but the pstrKey is for indexing

Reply:
yes, it was the invitee one I think needs removing